src/hid_freebsd.c

```diff
@@ -140,25 +123,20 @@ copy_info_hidraw(fido_dev_info_t *di, const char *path)
 		goto fail;
 
 	if (ioctl(fd, IOCTL_REQ(USB_GET_DEVICEINFO), &udi) == -1) {
-		if (ioctl(fd, IOCTL_REQ(HIDIOCGRAWINFO), &devinfo) == -1)
-			goto fail;
-
-		if (ioctl(fd, IOCTL_REQ(HIDIOCGRAWNAME(128)), rawname) == -1)
+		if (ioctl(fd, IOCTL_REQ(HIDIOCGRAWINFO), &devinfo) == -1 ||
+		    ioctl(fd, IOCTL_REQ(HIDIOCGRAWNAME(128)), rawname) == -1)
 			goto fail;
-
 		if ((di->path = strdup(path)) == NULL ||
 		    (di->manufacturer = strdup(UHID_VENDOR)) == NULL ||
 		    (di->product = strdup(rawname)) == NULL)
 			goto fail;
-
```

Comment:
If you don't like this empty line, maybe also remove it from copy_info_uhid for consistency.